IDeliverable.Utils.Core/Handlers/IHandlerExtensions.cs

```diff
@@ -15,7 +16,7 @@ public async static Task HandleAsync<TMessage>(this IEnumerable<IHandler<TMessag
             {
```

Comment:
Now that we have a `CancellationToken` maybe we should check it on each iteration here?

Reply:
I considered that, but thought it best that the handlers themselves were in charge of this